packages/ui/src/providers/keyring/provider.tsx

```diff
@@ -15,21 +15,46 @@ function isKeyringLoaded() {
   }
 }
 
+const loadKeysFromExtension = async () => {
+  await web3Enable('Pioneer')
+  const injectedAccounts = await web3Accounts()
+  const allAccounts = injectedAccounts.map(({ address, meta }) => ({
+    address,
+    meta: { ...meta, name: `${meta.name} (${meta.source})` },
+  }))
+
+  if (!isKeyringLoaded()) {
+    keyring.loadAll({ isDevelopment: true }, allAccounts)
+  }
+}
+
+// Extensions is not always ready on application load, hence the check
+const onExtensionLoaded = (callback: () => void) => () => {
+  const intervalId = setInterval(() => {
+    if (Object.keys((window as any).injectedWeb3).length) {
```

Comment:
Can this check be simplified perhaps?
```suggestion
    if ((window as any).injectedWeb3 !== undefined) {
```

Reply:
I'll double check but AFAIR the `window.injectedWeb3` object is available immediately and the `Object.keys()` heuristic is used to detect the moment when the extension is ready.  
  
I did not found anything better in the API :/

Reply:
Fair enough